rpc/blockchain.go

```diff
@@ -128,8 +150,22 @@ func (s *PublicBlockchainService) BlockNumber(ctx context.Context) (interface{},
 func (s *PublicBlockchainService) GetBlockByNumber(
 	ctx context.Context, blockNumber BlockNumber, opts interface{},
 ) (response StructuredResponse, err error) {
-	// Process arguments based on version
+
 	blockNum := blockNumber.EthBlockNumber()
+	if block, ok := s.blockCache.Get(uint64(blockNum)); ok {
+		return block.(StructuredResponse), nil
+	}
+
+	if s.limiter != nil {
```

Comment:
Can we refactor this into a function since this is heavily used in RPC calls?

Reply:
Yes, it will .